### ISSUE TITLE
Formatting namespaced maps with the native image

### ISSUE THREAD

**Issue by ash14** [kkinnear/zprint#104]
First of all thanks for making this.

I've noticed some unexpected behaviour when dealing with namespaced maps using the native image.

Consider this case:

```
$ clj -Sdeps '{:deps {zprint {:mvn/version "0.4.15"}}}'
Clojure 1.10.0
user=> (require '[zprint.core :refer :all])
nil
user=> (czprint {:a :b, :c #:c{:e :f :g        :h}})
{:a :b, :c #:c{:e :f, :g :h}}
```

Versus the native image:

```
$ ./zprintm-0.4.15 <<< '{:a :b, :c #:c{:e :f :g        :h}}'
{:a :b :c #:c{:e :f :g        :h}}
```

It appears that namespaced maps are ignored altogether.

**Comment by kkinnear** [comment]
Sorry it has taken me a bit longer than I would like to respond to this issue.  Also, thanks for the thanks, they tend to be in moderately short supply.

Let me start off by saying that I find namespaces maps rather confusing.  Additionally, the underlying parser used by zprint isn't altogether on board with them either.

That said, some preliminary testing suggests that this is due to a change in Clojure versions.  You are using 1.10, and it seems to work ok.  I've tried 1.9, and that appears to work the same.  In my 1.8 repl, I can't even parse these namespaces maps (i.e. `read-string` won't accept them).  The native-image zprint images use Clojure 1.8 (since they are built from the same uberjar I release for use with app-cds and direct use).  All of these use 1.8 since until recently leiningen plugins wouldn't work with anything later.  And lein-zprint uses the standard zprint.  Quite the chain of dependencies.

However, I believe that leiningen has moved to 1.9 some good while ago, and in order to fix this, I should move the entire ensemble to 1.9.  If I do, I expect that it would fix this native image issue, though until I actually see it work I won't be entirely confident about that. 

It will be a good many weeks at best before this might happen, as I'm tied up with a lot of other things as well as being in the middle of a major implementation of `:respect-nl?` for zprint.  I have been hoping to complete that implementation before the next release, but there are enough bugs out there to fix that I may have to do an intermediate release.  

Thanks for noticing this and reporting it.  I have a few namespaced key and map tests, but obviously not enough.

**Comment by ash14** [comment]
> Let me start off by saying that I find namespaces maps rather confusing.

Confusing maybe, I've certainly been hit when trying (and failing) to find certain keywords which namespaced maps turned out to be covering up.

Anyway, a workaround for this is to [disable printing namespaced maps](https://groups.google.com/forum/#!topic/clojure/XAP0ENMSUTc), so no rush with this.

**Comment by kkinnear** [comment]
I have actually fixed this in version 0.4.16.  I have no idea what i was thinking when I talked about needing to move to Clojure 1.9 for this.  I did do that move, but it had no effect on this issue.  This was a code problem that I put considerable effort into fixing.  It seems to work correctly now.  Thanks again for submitting this issue.
